release/man/guff.1
.TH "GUFF" "1" "October 2026" "guff v0.12.0" "User Commands"
.SH NAME
GUFF \- Manual page for guff v0.12.0.
.SH DESCRIPTION
A simple SASS/SCSS compiler and CSS parser/minifier.
.SS USAGE:
.TP
guff [FLAGS] [OPTIONS]
.SS FLAGS:
.TP
\fB\-e\fR, \fB\-\-expanded\fR
Do not minify CSS.
.TP
\fB\-h\fR, \fB\-\-help\fR
Print help information and exit.
.TP
\fB\-V\fR, \fB\-\-version\fR
Print version information and exit.
.SS OPTIONS:
.TP
\fB\-b\fR, \fB\-\-browsers\fR <VER>
A comma\-separated list of specific browser/version pairs to target for CSS compatibility, like 'firefox 90, ie 11'.
.TP
\fB\-i\fR, \fB\-\-input\fR <FILE>
The path to an SCSS or CSS source file.
.TP
\fB\-o\fR, \fB\-\-output\fR <FILE>
The path to save the minified output to. If omitted, the result will be printed to STDOUT instead.


guff_css/CHANGELOG.md
# Changelog


## [0.12.0](https://github.com/Blobfolio/guff/releases/tag/v0.12.0) - 2026-10-01

### Changed

* Bump MSRV to `1.95`
* Bump `trimothy` to `0.10`
* Update browser data


## [0.11.6](https://github.com/Blobfolio/guff/releases/tag/v0.11.6) - 2026-07-20

### Changed

* Bump `lightningcss` to `alpha72`
* Update browser data


## [0.11.5](https://github.com/Blobfolio/guff/releases/tag/v0.11.5) - 2026-07-09

### Changed

* Update browser data


## [0.11.4](https://github.com/Blobfolio/guff/releases/tag/v0.11.4) - 2026-04-16

### Changed

* Update browser data


## [0.11.3](https://github.com/Blobfolio/guff/releases/tag/v0.11.3) - 2026-03-20

### Changed

* Bump `lightningcss` to `alpha71`
* Update browser data


## [0.11.2](https://github.com/Blobfolio/guff/releases/tag/v0.11.2) - 2026-03-05

### Changed

* Update browser data


## [0.11.1](https://github.com/Blobfolio/guff/releases/tag/v0.11.1) - 2026-01-22

### Changed

* Bump `lightningcss` to `alpha70`


## [0.11.0](https://github.com/Blobfolio/guff/releases/tag/v0.11.0) - 2025-10-30

### Changed

* Bump MSRV to `1.90`
* Miscellaneous code changes and lints


## [0.10.2](https://github.com/Blobfolio/guff/releases/tag/v0.10.2) - 2025-10-06

### Changed

* Fix `docs.rs` generation


## [0.10.1](https://github.com/Blobfolio/guff/releases/tag/v0.10.1) - 2025-09-29

### New

* Re-export `grass`

### Changed

* Bump `lightningcss` to `alpha68`


## [0.10.0](https://github.com/Blobfolio/guff/releases/tag/v0.10.0) - 2025-09-18

### New

* Re-export `lightningcss`

### Changed

* Remove `dactyl` dependency
* Remove `minreq` dependency (build)
* Remove `oxford_join` dependency
* Remove `serde_json` dependency (build)
* Remove `serde` dependency (build)

### Breaking

* Remove `Agent`
* Remove `Agents`
* `Css::minified` now takes `Option<Browsers>` directly


## [0.9.0](https://github.com/Blobfolio/guff/releases/tag/v0.9.0) - 2025-06-26

### Changed

* Bump `dactyl` to `0.13`
* Bump `lightningcss` to `alpha67`
* Bump `oxford_join` to `0.7`
* Bump `trimothy` to `0.9`
* Bump MSRV to `1.88`
* Miscellaneous code changes and lints


## [0.8.1](https://github.com/Blobfolio/guff/releases/tag/v0.8.1) - 2025-06-01

### Changed

* Bump `dactyl` to `0.12`
* Bump `oxford_join` to `0.6`
* Bump `trimothy` to `0.8`
* Miscellaneous code changes and lints


## [0.8.0](https://github.com/Blobfolio/guff/releases/tag/v0.8.0) - 2025-05-30

### Changed

* Bump `dactyl` to `0.11`
* Bump MSRV to `1.87`
* Fix deps.rs badge


## [0.7.4](https://github.com/Blobfolio/guff/releases/tag/v0.7.4) - 2025-05-15

### Changed

* Bump `lightningcss` to `alpha66`
* Miscellaneous code changes and lints


## [0.7.3](https://github.com/Blobfolio/guff/releases/tag/v0.7.3) - 2025-04-03

### Changed

* Miscellaneous code changes and lints


## [0.7.2](https://github.com/Blobfolio/guff/releases/tag/v0.7.2) - 2025-03-14

### Changed

* Bump `lightningcss` to `alpha65`


## [0.7.1](https://github.com/Blobfolio/guff/releases/tag/v0.7.1) - 2025-03-06

### Changed

* Bump `lightningcss` to `alpha64`


## [0.7.0](https://github.com/Blobfolio/guff/releases/tag/v0.7.0) - 2025-02-25

### Changed

* Bump `dactyl` to `0.10`
* Bump `oxford_join` to `0.5`
* Bump `trimothy` to `0.7`
* Bump MSRV to `1.85`
* Bump Rust edition to `2024`
* Miscellaneous code changes and lints


## [0.6.1](https://github.com/Blobfolio/guff/releases/tag/v0.6.1) - 2025-02-20

### Changed

* Miscellaneous code changes and lints


## [0.6.0](https://github.com/Blobfolio/guff/releases/tag/v0.6.0) - 2025-01-09

### Changed

* Bump `dactyl` to `0.9`
* Bump `lightningcss` to `alpha63`
* Bump MSRV to `1.83`
* Miscellaneous code changes and lints


## [0.5.3](https://github.com/Blobfolio/guff/releases/tag/v0.5.3) - 2024-11-28

### Changed

* Bump `dactyl` to `0.8`
* Bump `lightningcss` to `alpha61`
* Bump `trimothy` to `0.6`


## [0.5.2](https://github.com/Blobfolio/guff/releases/tag/v0.5.2) - 2024-11-04

### Changed

* Bump `lightningcss` to `alpha60`


## [0.5.1](https://github.com/Blobfolio/guff/releases/tag/v0.5.1) - 2024-10-17

### Changed

* Bump `lightningcss` to `alpha59`
* Bump `oxford_join` to `0.3`
* Reduce intermediary string allocations during build


## [0.5.0](https://github.com/Blobfolio/guff/releases/tag/v0.5.0) - 2024-09-05

### Changed

* Bump MSRV to `1.81`
* Minor code lints


## [0.4.0](https://github.com/Blobfolio/guff/releases/tag/v0.4.0) - 2024-08-06

### Changed

* Bump MSRV to `1.80`
* Bump `grass` to `0.13.4`
* Bump `lightningcss` to `alpha58`
* Bump `trimothy` to `0.3`
* Bump `minreq` to `2.12` (build)


## [0.3.4](https://github.com/Blobfolio/guff/releases/tag/v0.3.4) - 2024-05-25

### Changed

* Bump `lightningcss` to `alpha57`


## [0.3.3](https://github.com/Blobfolio/guff/releases/tag/v0.3.3) - 2024-05-18

### Changed

* Bump `grass` to `0.13.3`
* Bump `lightningcss` to `alpha56`


## [0.3.2](https://github.com/Blobfolio/guff/releases/tag/v0.3.2) - 2024-03-15

### Changed

* Bump `lightningcss` to `alpha55`


## [0.3.1](https://github.com/Blobfolio/guff/releases/tag/v0.3.1) - 2024-02-23

### Changed

* Bump `lightningcss` to `alpha54`


## [0.3.0](https://github.com/Blobfolio/guff/releases/tag/v0.3.0) - 2024-02-14

### Changed

* Bump `lightningcss` to `alpha53`
* Bump MSRV to `1.72`


## [0.2.12](https://github.com/Blobfolio/guff/releases/tag/v0.2.12) - 2024-02-08

### Changed

* Replace `ureq` build dependency w/ `minreq`
* Bump `dactyl` to `0.7`


## [0.2.11](https://github.com/Blobfolio/guff/releases/tag/v0.2.11) - 2024-01-14

### Changed

* Bump `lightningcss` to `alpha52`


## [0.2.10](https://github.com/Blobfolio/guff/releases/tag/v0.2.10) - 2023-11-16

### Changed

* Bump `lightningcss` to `alpha51`


## [0.2.9](https://github.com/Blobfolio/guff/releases/tag/v0.2.9) - 2023-10-15

### Changed

* Bump `dactyl` to `0.6`
* Bump `lightningcss` to `alpha49`


## [0.2.8](https://github.com/Blobfolio/guff/releases/tag/v0.2.8) - 2023-10-05

### Changed

* Bump `trimothy` to `0.2`



release/completions/guff.bash
_basher___guff() {
	local cur prev opts
	COMPREPLY=()
	cur="${COMP_WORDS[COMP_CWORD]}"
	prev="${COMP_WORDS[COMP_CWORD-1]}"
	opts=()
	if [[ ! " ${COMP_LINE} " =~ " -e " ]] && [[ ! " ${COMP_LINE} " =~ " --expanded " ]]; then
		opts+=("-e")
		opts+=("--expanded")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -h " ]] && [[ ! " ${COMP_LINE} " =~ " --help " ]]; then
		opts+=("-h")
		opts+=("--help")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -V " ]] && [[ ! " ${COMP_LINE} " =~ " --version " ]]; then
		opts+=("-V")
		opts+=("--version")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -b " ]] && [[ ! " ${COMP_LINE} " =~ " --browsers " ]]; then
		opts+=("-b")
		opts+=("--browsers")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -i " ]] && [[ ! " ${COMP_LINE} " =~ " --input " ]]; then
		opts+=("-i")
		opts+=("--input")
	fi
	if [[ ! " ${COMP_LINE} " =~ " -o " ]] && [[ ! " ${COMP_LINE} " =~ " --output " ]]; then
		opts+=("-o")
		opts+=("--output")
	fi
	opts=" ${opts[@]} "
	if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
		COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
		return 0
	fi
	case "${prev}" in
		--input|--output|-i|-o)
			if [ -z "$( declare -f _filedir )" ]; then
				COMPREPLY=( $( compgen -f "${cur}" ) )
			else
				COMPREPLY=( $( _filedir ) )
			fi
			return 0
			;;
		*)
			COMPREPLY=()
			;;
	esac
	COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
	return 0
}
complete -F _basher___guff -o bashdefault -o default guff
